app/controllers/concerns/attachments_copier.rb

```diff
@@ -0,0 +1,20 @@
+module AttachmentsCopier
+  def copy_attachments(record)
+    record.content.scan(Attachment::SCREENSHOT_REGEX).each do |screenshot_path|
+      full_screenshot_path, _, _, _, project_id, node_id, filename, _ = screenshot_path
+
+      attachment = Attachment.find_by(filename: CGI::unescape(filename), node_id: record.node_id_was)
+
+      if attachment
+        new_attachment = attachment.copy_to(record.node)
+        new_filename = new_attachment.url_encoded_filename
+        new_path = full_screenshot_path.gsub(
+          /nodes\/[0-9]+\/attachments\/.+/,
+          "nodes/#{new_attachment.node_id}/attachments/#{new_filename}"
+        )
```

Comment:
I'm concerned about this text manipulation. Then I thought about the "copy link" feature of the attachments sidebar, perhaps we can use the same approach?

https://github.com/dradis/dradis-ce/blob/main/app/views/attachments/_attachment_box.html.erb#L59

I could even consider moving this into a helper to avoid duplication, but we don't currently have an attachment_helper.rb it could fit in.

Reply:
It's tricky to DRY this considering that the contexts are different. The attachment box is in the view that has access to the route helpers while this code isn't. There's also some possible rabbit hole with the `default_url_options = /pro` once we move the link helpers out of the view. Happy to explore this more if we need this code on many more places but currently, this seems like the simplest approach.